run_project_tests.py

```diff
@@ -47,6 +47,12 @@
 from run_tests import ensure_backend_detects_changes
 from run_tests import guess_backend
 
+ALL_TESTS = ['cmake', 'common', 'warning-meson', 'failing-meson', 'failing-build', 'failing-test',
+             'kconfig', 'platform-osx', 'platform-windows', 'osx', 'windows', 'platform-linux',
+             'linuxlike', 'java', 'C#', 'vala',  'rust', 'd', 'objective c', 'objective c++',
+             'fortran', 'swift', 'cuda', 'python3', 'python', 'fpga', 'frameworks', 'nasm', 'wasm'
+             ]
+
```

Comment:
This Duplicates & Diverges `all_tests` in `detect_tests_to_run()`. I guess `all_tests` is generated much later? So maybe you can at least block the divergence by adding something like this in `detect_tests_to_run()`:
```
assert( set (ALL_TESTS) == set([ tst[1] for tst in all_tests ])  )
```

Reply:
Maintaining these things by hand means they will always be out of date. At the very least it should build this by globbing the test dir.

Reply:
The directory names are different than some test names

Reply:
I also think **unless changing the test names**, that doing the assertion above would be what I would do.